taskworld/typescript-custom-transformer-l10n
# feat!: upgrade TypeScript

### Problems

TypeScript v5 comes with [API breaking changes](https://github.com/microsoft/TypeScript/wiki/API-Breaking-Changes#:~:text=The%20top%2Dlevel%20Node%20factory%20functions). Mostly `ts.create...` functions have been moved to `transformationContext.factory` (alternatively `ts.factory` namespace) and some are also gets their names changed: from `updateCall` to `updateCallExpression` for example.

The logic and the test suite are intact.

This PR also...
- [x] upgrades Jest and Prettier to the latest.
- [x] converts JS to TS files.
- [x] adds deployment automation (`npm publish`) via **GitHub Actions**.
- [x] fixes vulnerabilities from the older versions of `minimatch` and `lodash`.

### Steps to verify the solutions

Once this PR is merged, GitHub Actions will publish version 5.0.0 of this package to [GitHub package registry](https://github.com/taskworld/typescript-custom-transformer-l10n/pkgs/npm/typescript-custom-transformer-l10n).

1. Upgrade TypeScript to v5 in **tw-frontend** first by checking out [this PR](https://github.com/taskworld/tw-frontend/compare/master...manta/ts5)
2. Run `yarn build` on this PR.
3. Rename `index.js` to `transformer.js` and replace the existing _tw-frontend/node_modules/@taskworld/typescript-custom-transformer-l10n/transformer.js_
4. Expect no errors coming from `cd client && yarn build:standalone`
